crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractCustomResourceHandler.java

```diff
@@ -91,6 +98,54 @@ public void handle(CustomResourceInfo config) {
     });
   }
 
+  private TypeDef visitTypeDef(TypeDef def, List<Visitor<TypeDefBuilder>> visitors) {
+    if (parallel) {
+      return visitTypeDefInParallel(def, visitors);
+    } else {
+      return visitTypeDefSequentially(def, visitors);
+    }
+  }
+
+  private TypeDef visitTypeDefSequentially(TypeDef def, List<Visitor<TypeDefBuilder>> visitors) {
+    TypeDefBuilder builder = new TypeDefBuilder(def);
+    for (Visitor<TypeDefBuilder> visitor : visitors) {
+      builder.accept(visitor);
+    }
+    return builder.build();
+  }
+
+  private TypeDef visitTypeDefInParallel(TypeDef def, List<Visitor<TypeDefBuilder>> visitors) {
+    if (visitors.isEmpty()) {
+      return def;
+    }
```

Comment:
Perhaps move this `if` to the `visitTypeDef` method since it is useful for `visitTypeDefSequentially` too?

Reply:
done